src/include/interval_set.h

```diff
@@ -34,6 +34,9 @@ template<typename T, typename Map = std::map<T,T>>
 class interval_set {
  public:
   using value_type = T;
+  using map_reference = typename Map::reference;
```

Comment:
could you take this opportunity to fix the value_type? as the value_type here is actually `std::pair<const T, T>`.

Reply:
I don't understand. `value_type` is `T` which in most cases is `uint64_t`?

Reply:
i mean, this context, `value_type` is actually an interval, which can be represented by `std::pair<const T, T>`, and it's also the type of `*iterator`. it's also expected by users of this container.

Reply:
I like keeping `value_type = T` as it can be used elsewhere in the header. I've changed the name of `map_reference` to `interval_type`. What do you think of that?

Reply:
please note `value_type` is the type of `*iterator`, i think that's the convention of STL containers. if we use it as an alias of `T`, it's our fault.